src/irmin-pack/inode.ml

```diff
@@ -308,9 +314,11 @@ struct
 
     type _ layout =
       | Total : total_ptr layout
-      | Partial : (hash -> partial_ptr t option) -> partial_ptr layout
+      | Partial : find -> partial_ptr layout
       | Truncated : truncated_ptr layout
 
+    and find = depth:int -> hash -> partial_ptr t option
```

Comment:
Wouldn't a move descriptive name be better for that parameter? Like `expected_depth`